torch/lib/c10d/ProcessGroupGloo.cpp

```diff
@@ -786,34 +797,47 @@ class AsyncSparseAllreduceWork : public ProcessGroupGloo::AsyncWork {
   std::vector<at::Tensor> allgather_values(
       const at::Tensor& tensor,
       const std::vector<SparseTensorMetadata>& metadata) {
-    auto max_nnz = metadata[0].nnz();
-    for (size_t i = 1; i < metadata.size(); i++) {
-      max_nnz = std::max(max_nnz, metadata[i].nnz());
+    // There are nnz #dense_dim()-dimensional tensors per rank.
+    const auto valueShape = tensor.sizes().slice(tensor.sparse_dim());
+    size_t denseDims = 1;
```

Comment:
This is not the number of dimensions but the number of elements in the dense dimensions, so wouldn't `denseNumel` (or similar) be more descriptive here?